app/services/battle_service.rb

```diff
@@ -162,6 +162,41 @@ def unready_player(battle_id)
     end
   end
 
+  def kick_player(battle_id, kick_player_id)
+    # Validate caller is admin
+    raise BattleValidationError.new "Player #{@player} is not an admin" unless @player.admin?
+
+    # Validate battle exists
+    battle = validate_battle(battle_id)
+
+    battle.with_lock do
+
+      # Validate the player is in the battle
+      validate_player_in_battle(battle)
```

Comment:
Shouldn't this validate that the player to kick is in the battle? I don't actually think we care if the kicker is as it could be an admin action separate from the battle.